File: app_mobile_legacy/src/main/java/com/kee0kai/thekey/engine/model/DecryptedPassw.java
package com.kee0kai.thekey.engine.model;


import com.github.klee0kai.hummus.model.ICloneable;
import com.github.klee0kai.hummus.model.ISameModel;

import java.util.Objects;

public class DecryptedPassw implements ICloneable, ISameModel {

    public String passw;
    public long chTime;

    public DecryptedPassw(String passw, long chTime) {
        this.passw = passw;
        this.chTime = chTime;
    }

    @Override
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }

    @Override
    public boolean isSame(Object ob) {
        return ob instanceof DecryptedPassw && Objects.equals(passw, ((DecryptedPassw) ob).passw);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DecryptedPassw that = (DecryptedPassw) o;
        return chTime == that.chTime && Objects.equals(passw, that.passw);
    }

    @Override
    public int hashCode() {
        return Objects.hash(passw, chTime);
    }
}

File: app_mobile/src/main/kotlin/com/github/klee0kai/thekey/app/engine/model/DecryptedPassw.kt
package com.github.klee0kai.thekey.app.engine.model

import android.os.Parcelable
import com.github.klee0kai.brooklyn.JniPojo
import com.github.klee0kai.thekey.core.domain.model.HistPassw
import kotlinx.parcelize.Parcelize
import java.util.concurrent.TimeUnit

@JniPojo
@Parcelize
data class DecryptedPassw(
    val passwPtr: Long = 0,
    val passw: String = "",
    /**
     * change tim in seconds,
     */
    val chTime: Long = 0,
) : Parcelable


fun DecryptedPassw.histPasww(
    isLoaded: Boolean = false,
) = HistPassw(
    passwPtr = passwPtr,
    passw = passw,
    chTime = TimeUnit.SECONDS.toMillis(chTime),
    isLoaded = isLoaded,
)